Gems/Multiplayer/Code/Source/Components/LocalPredictionPlayerInputComponent.cpp

```diff
@@ -188,20 +189,25 @@ namespace Multiplayer
             return;
         }
 
+        // After receiving the first input from the client, start the update event to check for slow hacking.
+        // Also initialize the lastClientInputId to one before the oldest available one in the inputArray so that
+        // we process everything available to us on the first call.
+        if (!m_updateBankedTimeEvent.IsScheduled())
+        {
+            // This subtraction intentionally wraps around.
+            m_lastClientInputId = inputArray[NetworkInputArray::MaxElements - 1].GetClientInputId() - ClientInputId(1);
```

Comment:
I forget, is this input array always flushed out with events..? What do we do on the very first few inputs?

Reply:
Double checked, we flush with the oldest element, so what you have works great 👍

Reply:
Yeah, it works fine the way it's written, I'm just not personally enamored of it because it's continuing to make an assumption that the API doesn't really enforce, that all entries are valid. For example, we've got unit tests that do _not_ fill out all the elements but also don't fail. I'd rather see this code stop making the assumption, either by using a fixed_vector instead of an array and checking .size(), or adding flags to say if a given entry is valid, or worst-case relying on the host frame id being initialized to invalid to detect it.

But none of this is required for this bugfix, so I'm leaving it as-is for now.